### PR TITLE
Fix llorg/sycl-web pulldown issue

### PR DESCRIPTION
**Pull request by zahiraam**
Signed-off-by: Zahira Ammarguellat <zahira.ammarguellat@intel.com>

### CODE REVIEW

File: clang/include/clang/AST/DeclCXX.h [hunk 1]
```diff
@@ -1139,6 +1139,9 @@ class CXXRecordDecl : public RecordDecl {
   /// \note This does NOT include a check for union-ness.
   bool isEmpty() const { return data().Empty; }
 
+  void setInitMethod(bool Val) { data().HasInitMethod = Val; }
```
Comment (smanna12) [inline clang/include/clang/AST/DeclCXX.h:1142]: Could you please add a comment here?
Example: https://github.com/intel/llvm/blob/4d42a5c2c5eb3d77f720cb298691e165e768079e/clang/include/clang/AST/DeclCXX.h#L111

**Comment by Fznamznon** [comment]
Shouldn't the fix be included into https://github.com/intel/llvm/pull/5427 . @vmaksimo WDYT?

**Comment by zahiraam** [comment]
> Shouldn't the fix be included into #5427 . @vmaksimo WDYT?

I think it should. But I am wondering if better yet, the changes to SemaDecl.cpp shouldn't be up-streamed? And only make the change here for accessor.hpp?

> Shouldn't the fix be included into #5427 . @vmaksimo WDYT?

**Comment by Fznamznon** [comment]
> I am wondering if better yet, the changes to SemaDecl.cpp shouldn't be up-streamed?

I'm sure that the check on `NewFD->getIdentifier()` should be upstreamed. About the check on `SYCLIsDevice` I'm not that sure, since the attribute doc doesn't say that it shouldn't be used only for device compilation. It is our headers for some reason put some `__init` methods under `SYCL_DEVICE_ONLY` macro.
Probably it makes sense to commit the fix together with pulldown as well to unblock pulldown.

**Comment by zahiraam** [comment]
> > I am wondering if better yet, the changes to SemaDecl.cpp shouldn't be up-streamed?
> 
> I'm sure that the check on `NewFD->getIdentifier()` should be upstreamed. About the check on `SYCLIsDevice` I'm not that sure, since the attribute doc doesn't say that it shouldn't be used only for device compilation. It is our headers for some reason put some `__init` methods under `SYCL_DEVICE_ONLY` macro. Probably it makes sense to commit the fix together with pulldown as well to unblock pulldown.

I will make the change for NewFD->getIdentifier() in llorg and ask you to review it. To minimize the next pulldown "headaches" :)

**Comment by zahiraam** [comment]
> > > I am wondering if better yet, the changes to SemaDecl.cpp shouldn't be up-streamed?
> > 
> > 
> > I'm sure that the check on `NewFD->getIdentifier()` should be upstreamed. About the check on `SYCLIsDevice` I'm not that sure, since the attribute doc doesn't say that it shouldn't be used only for device compilation. It is our headers for some reason put some `__init` methods under `SYCL_DEVICE_ONLY` macro. Probably it makes sense to commit the fix together with pulldown as well to unblock pulldown.
> 
> I will make the change for NewFD->getIdentifier() in llorg and ask you to review it. To minimize the next pulldown "headaches" :)

And the rest of the changes can just be committed here. Thanks.

**Comment by vmaksimo** [comment]
I see it's the best solution - to commit the fix as a part of #5427 and to upstream the needed fix to llorg to minimize next week's pulldown problems. Thanks!

**Pull request closed without merging**